### PR TITLE
doc: various corrections and additions

### PR DESCRIPTION
**Pull request by adrienlucbert**
Made a few corrections/changes/additions that I deem necessary, but all of them remain open to discussion and can change.

I wanted to add about:
- Class naming convention (PascalCase)
- Abstract classes prefix with `A`
- Usage of `this.`
- DRY / KISS / SOLID
- a few examples ? keep it simple but a bit more orienting ?

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -28,13 +29,13 @@ reading through another contributor's code.
 
 ### Types of Naming Conventions
 
-| Common Name   | Example                                        | Notes  |
-| ------------- | ---------------------------------------------- | ------ |
-| Pascal-case   | `DesertEagle`                                  |        |
-| Camel-case    | `desertEagle`                                  |        |
-| Snake-case    | `desert_eagle`, `Desert_Eagle`, `desert_Eagle` |        |
```
Comment (adrienlucbert) [inline README.md:35]: [Wikipedia](https://en.wikipedia.org/wiki/Snake_case) states that *"the first letter of each word written in lowercase".*

File: README.md [hunk 2]
```diff
@@ -83,71 +86,125 @@ There are many common rules for naming and maintaining **Classes** within
...
 /// Fire a gun
 /// </summary>
-public void Fire()
+/// <param name="bulletType">Type of the bullet to fire</param>
+/// <returns>True if the gun was able to fire, false if it wasn't for any reason</returns>
+public bool Fire(BulletType bulletType)
 {
-// Fire the gun.
+  // Fire the gun.
 }
```
Comment (adrienlucbert) [inline README.md:111]: If that is too much, we can shorten the example, I mainly wanted to talk about the `xmldoc` format in addition to the `<summary>`  tag.

File: README.md [hunk 3]
```diff
@@ -83,71 +86,125 @@ There are many common rules for naming and maintaining **Classes** within
...
-There are many common rules for naming and maintaining **Functions** and **Event Handlers** within **Classes**.
+There are many common rules for naming and maintaining **Functions** and **Event
+Handlers** within **Classes**.
+
 - All function names should start with verbs
 - All function names should be worded in present tense
 - All function names should have context as to what they are doing
 - All event handler names should start with `On`
 
+### Design Principles
```
Comment (adrienlucbert) [inline README.md:202]: Again that may be a bit much, but I feel the need to include them, at least not as hard rules that people **must** follow, but more as guidelines that people **should** follow as much as possible.

File: README.md (hunk 1 above)
Comment (NAkbulut) [reply README.md:35]: Sourcing Wikipedia gives me the shivers, but I admit my mistake.

File: README.md (hunk 3 above)
Comment (NAkbulut) [reply README.md:202]: I think including design guidelines in the project conventions are not necessary, but a nice addition either way. I appreciate you adding them!

File: README.md (hunk 2 above)
Comment (NAkbulut) [reply README.md:111]: I'm OK with this.

File: README.md [hunk 4]
```diff
@@ -83,71 +86,125 @@ There are many common rules for naming and maintaining **Classes** within
...
 Use **Namespaces** to ensure your scoping of items won't conflict with existing
 ones from other- or global namespaces. The project should at minimum use the
 project name for its namespaces in order to prevent conflicts with imported
 third party assets.
 
-All **Public Functions** should have a summary.
-```
+All **Public Functions** should have documentation in [the xmldoc format](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/xmldoc/recommended-tags),
+containing at least a summary and a description of their parameters and return
+value if any.
```
Comment (NAkbulut) [inline README.md]:
```suggestion
value, if any.
```

File: README.md [hunk 5]
```diff
@@ -83,71 +86,125 @@ There are many common rules for naming and maintaining **Classes** within
...
 - Comment text should end with a period
 - Insert one space between the comment delimiter and the comment text
 
-For **Spacing**, use a single space after a comma between function arguments.
+Example:
+```csharp
+/// This is a comment.
+```
+
+For **Spacing**, use a single space after a comma.
```
Comment (Neehier) [inline README.md:128]:
```suggestion
For **Spacing**, use a single space after a comma between function arguments.
```

File: README.md [hunk 6]
```diff
@@ -83,71 +86,125 @@ There are many common rules for naming and maintaining **Classes** within
...
 
 Example: `Console.In.Read(myChar, 0, 1);`
+
 - Do not use a space after the parenthesis and function arguments
 - Do not use spaces between a function name and parenthesis
 - Do not use spaces inside brackets
 
+Class members should be accessed through the `this` keyword, to help tracking the
```
Comment (NAkbulut) [inline README.md]:
```suggestion
Class members should be accessed through the `this` keyword, as this helps with tracking the

```

File: README.md [hunk 7]
```diff
@@ -83,71 +86,125 @@ There are many common rules for naming and maintaining **Classes** within
...
   variables.
-- Always use [Pascal-case](#types-of-naming-conventions) for interfaces
+- Always use [Pascal-case](#types-of-naming-conventions) for interfaces and
+  classes
 - Always prefix private variables with an underscore
-- Always prefix boolean variables with a verb
-- Always prefix interfaces with an `I`
-- All non-boolean variable names must be clear, unambiguous, and descriptive
-  **nouns**
+- Always prefix boolean variables with a verb such as `is`, `has`, ...
```
Comment (NAkbulut) [inline README.md]:
```suggestion
- Always prefix boolean variables with a verb such as `is` or `has`.

```

File: README.md [hunk 8]
```diff
@@ -83,71 +86,125 @@ There are many common rules for naming and maintaining **Classes** within
...
+- All variable names must be clear, unambiguous, and descriptive **nouns**
 - Variable names must not be redundant with their context
+- Local variable and function parameter names should not conflict with other
```
Comment (NAkbulut) [inline README.md]:
```suggestion
- Local variable- and function parameter names should not conflict with other

```

File: README.md [hunk 9]
```diff
@@ -83,71 +86,125 @@ There are many common rules for naming and maintaining **Classes** within
...
+  variables in the same scope (class variables, global variables, ...)
```
Comment (NAkbulut) [inline README.md]:
```suggestion
  variables within the same scope (such as class variables, global variables, etc.)

```

**Review by NAkbulut — CHANGES_REQUESTED** [review]

File: README.md [hunk 10]
```diff
@@ -62,15 +64,16 @@ We use styling conventions which conform to **Microsoft's C# Standard**.
 
 There are many common rules for naming and maintaining **Classes** within
 **Source Files**.
-- Source files should contain only one `public` class
+
+- One source file should contain only one `public` class
```
Comment (NAkbulut) [inline README.md]:
```suggestion
- Each source file should contain only one `public` class

```

File: README.md [hunk 11]
```diff
@@ -83,71 +86,148 @@ There are many common rules for naming and maintaining **Classes** within
...
+
+For **Spacing**, use a single space after a comma.
 
 Example: `Console.In.Read(myChar, 0, 1);`
+
 - Do not use a space after the parenthesis and function arguments
 - Do not use spaces between a function name and parenthesis
 - Do not use spaces inside brackets
 
+Write variables- and functions **modifiers** in that order:
```
Comment (NAkbulut) [inline README.md]:
```suggestion
Variable- and function **modifiers** should be sorted in the following order:

```

**Review by NAkbulut — APPROVED** [review]

**Pull request merged**